src/main/scala/org/broadinstitute/dsde/workbench/sam/service/StatusService.scala

```diff
@@ -18,6 +17,7 @@ import scala.concurrent.{ExecutionContext, Future}
 
 class StatusService(
     val directoryDAO: DirectoryDAO,
+    val registrationDAO: RegistrationDAO,
```

Comment:
Wondering if we shouldn't just be explicit here and say:

```suggestion
    val ldapRegistrationDAO: LdapRegistrationDAO,
```

For this class, we don't want a generic `RegistrationDAO`, we explicitly need it to be an `LdapRegistrationDAO` in order to satisfy the business requirement.  This will require a little refactoring in Boot.scala too I think, but I think it would be worth it.  As it is here, I had to trace all the calls back through Boot.scala just to make sure we were passing in the right kind of `RegistrationDAO` here, which was annoying in itself, and also doesn't prevent anyone from accidentally breaking this again in the future. 

I realize this will mess with tests where we use `MockRegistrationDAO`.  I don't know the answer off the top of my head and maybe it's something we can sort out with a discussion/mob/pair to see if we can make this better in some way.

Reply:
We don't actually have a `MockRegistrationDAO`, interestingly enough! That said, I think changing it assumes that LDAP is how we communicate to our registration service. I might rather have it be `checkRegistrationDBStatus` rather than `checkOpenDJStatus`? Given that things changed with Postgres, I don't want to assume LDAP will be the way forwards forever.

Reply:
Agreed that we typically don't want to assume that LDAP will be the way forwards forever (I hope the RegistrationDAO goes away entirely one day). However, in this case, I'd argue that we want the `StatusService` checks to be implementation-specific. Keeping it generic was how we missed this in the first place and didn't check OpenDJ's status for >1 year. If we change the implementation of the RegistrationDAO, we'd want to be forced to change the `StatusService` to check the new implementation of it so that we couldn't accidentally leave it checking OpenDJ

Reply:
Ack. That seems reasonable in rationale to me. I will make the change.

Reply:
I went down a rabbit hole on this, specifying it as `LDAPRegistrationDao` makes mocking it exceptionally hard, because SBT/Mockito doesn't know how to override a class, only the traits that we use. So the best we could do is spy on it and then override the spy. This is going to be super clunky across our tests. I think the best thing we can do is to rename the variable and make a comment, which will be both simpler, and also provide a check for us against future changes.